### PR TITLE
Feature/api development

### PR DESCRIPTION
**Pull request by FurkanKayar55**
## Describe this PR

Residential Care Package
Nursing Care Package
Naming and standards

### *What changes have we introduced*

Add endpoints for Residential Care Package
Add endpoints for Nursing Care Package
change endpoints routes name

### CODE REVIEW

File: LBH.AdultSocialCare.Api/V1/Infrastructure/Entities/HomeCarePackage.cs [hunk 1]
```diff
@@ -14,16 +14,6 @@ public class HomeCarePackage
         [JsonProperty("Id")]
```
Comment (petemade) [inline LBH.AdultSocialCare.Api/V1/Infrastructure/Entities/HomeCarePackage.cs:14]: The db entities will not need JSON attributes

**Review by petemade — CHANGES_REQUESTED** [review]

File: LBH.AdultSocialCare.Api/V1/Gateways/Interfaces/IResidentialCarePackageGateway.cs [hunk 2]
```diff
@@ -9,5 +9,7 @@ namespace LBH.AdultSocialCare.Api.V1.Gateways.Interfaces
     public interface IResidentialCarePackageGateway
     {
         public Task<ResidentialCarePackage> UpsertAsync(ResidentialCarePackage residentialCarePackage);
+
+        public Task<ResidentialCarePackage> GetAsync(Guid residentialCarePackageId);
```
Comment (dunnudge) [inline LBH.AdultSocialCare.Api/V1/Gateways/Interfaces/IResidentialCarePackageGateway.cs]: More descriptive names for functions here too. `UpsertResidentialCarePackageAsync()`, `GetResidentialCarePackageAsync()`

File: LBH.AdultSocialCare.Api/V1/UseCase/UpsertNursingCarePackageUseCase.cs [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        public async Task<NursingCarePackageDomain> ExecuteAsync(NursingCarePackageDomain nursingCarePackage)
+        {
+            var nursingCarePackageEntity = NursingCarePackageFactory.ToEntity(nursingCarePackage);
+            nursingCarePackageEntity = await _gateway.UpsertAsync(nursingCarePackageEntity).ConfigureAwait(false);
+            if (nursingCarePackageEntity == null) return nursingCarePackage = null;
+            else
+            {
+                nursingCarePackage = NursingCarePackageFactory.ToDomain(nursingCarePackageEntity);
+            }
+            return nursingCarePackage;
```
Comment (dunnudge) [inline LBH.AdultSocialCare.Api/V1/UseCase/UpsertNursingCarePackageUseCase.cs:29]: In place of returning null and adding logic to handle it in the controller, you can throw an `EntityNotFoundException` that is handled in the global middleware. Just a recommendation. Do this after all the merges later this error handling update is on another feature branch.

**Review by dunnudge — APPROVED** [review]
Not exactly errors noted anywhere but some recommendations

File: LBH.AdultSocialCare.Api/V1/Boundary/Request/NursingCarePackageRequest.cs [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+        /// <summary>
+        /// Gets or sets the Client Id
+        /// </summary>
+        public Guid ClientId { get; set; }
+
+        /// <summary>
+        /// Gets or sets the Start Date
+        /// </summary>
+        public DateTime StartDate { get; set; }
```
[the remark this answers is not in the document]
Comment (FurkanKayar55) [reply LBH.AdultSocialCare.Api/V1/Boundary/Request/NursingCarePackageRequest.cs:23]: My bad, thanks!

File: LBH.AdultSocialCare.Api/V1/Controllers/ResidentialCarePackageController.cs [hunk 5]
```diff
@@ -39,5 +42,22 @@ public async Task<ActionResult<ResidentialCarePackageResponse>> Create(Residenti
                 return BadRequest(ex.Message);
             }
         }
+
+        [HttpGet]
+        [Route("/{residentialCarePackageId}")]
```
[the remark this answers is not in the document]
Comment (FurkanKayar55) [reply LBH.AdultSocialCare.Api/V1/Controllers/ResidentialCarePackageController.cs:47]: Thank you mate!

File: LBH.AdultSocialCare.Api/V1/Controllers/ServiceController.cs [hunk 6]
```diff
@@ -66,7 +65,6 @@ public async Task<ActionResult<ServiceResponse>> Get(Guid serviceId)
         }
 
         [HttpGet]
-        [Route("getAll")]
         public async Task<ActionResult<IList<PackageServices>>> GetAll()
         {
```
[the remark this answers is not in the document]
Comment (FurkanKayar55) [reply LBH.AdultSocialCare.Api/V1/Controllers/ServiceController.cs:69]: Makes sense.

**Pull request closed without merging**